Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntitySpawnListener.java

```diff
@@ -75,8 +75,7 @@ public static void testNether(final Entity entity) {
 
     public static void testCreate(final Entity entity) {
         @NonNull World world = entity.getWorld();
-        if (areaName.equals(world.getName())) {
```

Comment:
Should probably use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object- here

Reply:
I think I've a state where  Objects.equals doesn't work.

Think of areaName != null  but World#getName() is null. -> that results in a not equals.
-> it enters the if branch.

`
areaName = world.getName(); // will be null then.
hasPlotArea = PlotSquared.get().getPlotAreaManager().hasPlotArea(areaName); // will throw an exception because the parameter is marked as nonNull.

I assume that we don't want that case to be happen? Did I miss something?


`

Reply:
According to [the documentation](https://papermc.io/javadocs/paper/1.18/org/bukkit/generator/WorldInfo.html#getName()), `World#getName()` will never return `null`. So just using `world.getName().equals(areaName)` should be sufficient too, with no null-check needed

Reply:
See 0869565